soc/nxp/imx/imx8ulp/Kconfig.soc

```
@@ -13,7 +13,7 @@ config SOC_MIMX8ULP
 	select SOC_SERIES_IMX8ULP
 
 config SOC
-	default "imx8ulp" if SOC_MIMX8ULP
+	default "mimx8ud7" if SOC_MIMX8ULP
```

```suggestion
	default "mimx8ud7" if SOC_MIMX8UD7
```

We need to rename `SOC_MIMX8ULP` to `SOC_MIMX8UD7` elsewhere as well

---

@danieldegrasse Is this mandatory? I believe not.
We can name the `CONFIG_SOC_XYX` as suggestive as possible, and `SOC_MIMX8ULP` is more appropriate then `SOC_MIMX8UD7`, IMO. (8ULP is the board name).
I agree we have to have the right value set for it - in this case `mimx8ud7` (set correctly here), to link it with our NXP_HAL.

Is this CONFIG_SOC_XYX somewhere parsed and the "XYZ" is used?
IIUC only the value of CONFIG_SOC_XYX is used.

PS: We did the same for 8MPlus - https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/nxp/imx/imx8m/Kconfig.soc#L27 or others (8QXP or 8QM).

---

I'm going off (my understanding) of the guidance for HWMv2, that the CONFIG_SOC_xxx suffix needs to match the SOC name in the YAML- @nordicjm, is this correct?

---

My 2 cents here: switching to SOC-based (e.g: for ULP use `MIMX8UD7` and so on) naming scheme should be addressed in another patch series (if required, of course). This is because:

1. We have multiple SoCs that don't abide by this rule (there's also MP, QM, and QXP as @iuliana-prodan pointed out). Doing this in a separate patch with its only scope being fixing such irregularities should be more "clean" and easy to follow.
2. Doing this change also implies making even more changes on SOF side, which is not ideal. We should strive to make such changes incrementally to minimize the chance of breaking stuff. We're also in the process of transitioning to Zephyr drivers with a couple of patches that are still in review due to some Zephyr patches still being in review and having to also take the renaming into account would just make the process more prone to errors.
3. The scope of this PR is to just add the HAL binding. Having to modify the SOC name is just a side effect of this, not the main goal. The old `imx8ulp` name has just been a "placeholder" since we didn't need the HAL binding up until now so this was long overdue.

---

Keys and values for SOC, SOC_SERIES and SOC_FAMILY need to match (SOC can additionally contain the CPU cluster). Whilst this is not enforced by CI at the present time, it needs to be followed as in future lots of these Kconfig options will be generated automatically by the build system and at that point mismatches will suddenly cause entire SoCs to just break.

---

@nordicjm  @danieldegrasse can  we leave this as is now (as @LaurentiuM1234  suggested [here](https://github.com/zephyrproject-rtos/zephyr/pull/70219#discussion_r1532746679)) and I'll follow up with a PR next week?

@nordicjm I want to understand what does it have to match exactly?

If my [SOC value is `mimx8ml8`](https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/nxp/imx/imx8m/Kconfig.soc#L76), the key for [SOC_IMX8MP](https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/nxp/imx/imx8m/Kconfig.soc#L27) should be SOC_MIMX8ML8? 
Why is this mandatory? Since the SOC_IMX8MP is just a bool - is defined or not.

Also, what should match in the SOC_SERIES or SOC_FAMILY?
What is wrong here, for example?
```
config SOC_SERIES
	default "imx8m" if SOC_SERIES_IMX8M
```
or
```
config SOC_FAMILY
	default "nxp_imx" if SOC_FAMILY_NXP_IMX
```
Thanks!

---

No it needs to be updated here.
For:
```
config SOC_SERIES
	default "imx8m" if SOC_SERIES_IMX8M
```
then the `imx8m` part must match in the format `SOC_SERIES_<part>`. For:
```
config SOC_FAMILY
	default "nxp_imx" if SOC_FAMILY_NXP_IMX
```
then the `nxp_imx` part must match in the format `SOC_FAMILY_<family>`. For:
```
config SOC
	default "mimx8ud7" if SOC_MIMX8ULP
```
then the `mimx8ud7` part must match in the format `SOC_FAMILY_<soc>` or `SOC_FAMILY_<soc>_<cpucluster>`, so this is invalid and needs fixing